propulsion-hotel/Reactor.Integration/ReactorIntegrationTests.fs

```diff
@@ -0,0 +1,84 @@
+module Reactor.Integration.ReactorIntegrationTests
+
+open Domain
+open FsCheck
+open FsCheck.Xunit
+open System
+
+let runCheckoutScenario store (paymentId, id, NonEmptyArray stays, payBefore) check = async {
+    let staysService = GuestStay.Config.create store
+    let checkoutService = GroupCheckout.Config.create store            
+    let mutable charged = 0
+    for stayId, chargeId, PositiveInt amount in stays do
+        charged <- charged + amount
+        do! staysService.Charge(stayId, chargeId, amount)
+    let stays = [| for stayId, _, _ in stays -> stayId |]
+    if payBefore then do! checkoutService.Pay(id, paymentId, charged)
+    let! _ = checkoutService.Merge(id, stays)
+    do! check "awaiting Group Checkout action" <| fun wait -> async {
```

Comment:
It's not really clear what this `check` is. I traced it down with some effort to see that it's a partially `ReactorMonitor.check` from the propulsion MemoryStore.

Depending on the goal of this project, it might be wise to drop the Dynamo wiring from the tests so they can be scanned more easily

Reply:
Hard to say what the goal of this project is as such, but the threads for me are....
a) there should be a maintained template kept in sync that has wiring for a >=1 concrete store AND a memory store
b) too many concrete stores is too much noise (maybe a more thorough one with all the stores could go in Oskar's EventSourcign.NetCore)
c) ideally there should be an article with an associated uber-commented impl somewhere. That could remove the concrete store, and/or use one that has less noisy wiring (e.g. dockerized MessageDb). That also allows the template version to have less noisy/complete comments that one could actually leave in a real codebase and/or not bother people who have been through a Tutorial. Either that goes in Equinox/samples/Tutorial, or in a third party repo ...

Anyway, the other thing that I'll try to add to the `check` comments is that the idea is that any span of logic in an integration test that relies on reactions which may have a non-deterministic element (and hence sometimes require retries if the assertion is not iniitally true) gets wrapped in this check and retry loop.

In the code from which this was extracted, `check` was a decent name as there were multiple phases of processing with obvious grouping and/or need for potential retries. Will see if I can come up with a better name in this context.

(The other point is that with MemoryStore and MemoryStoreProjector, you can wait for all the events that have been raised to have been deemed handled by the projector and then move on. For a concrete store such as DynamoStore, there is no real way to know when you're definitely done in the same way)

Reply:
renamed to `checkWithRetry` and added some comments...